### DISCUSSION TITLE
Pybamm for V2G battery degradation

### DISCUSSION THREAD

**Discussion by iurkens** [pybamm-team/PyBaMM#2377] [Q&A] [answered]
Hello

I want to simulate battery degradation caused by various V2G applications. I'm an electrical engineering student, so my focus is only partially on electrochemistry.

I want to add algorithms that work with load profiles and optimize DoD, peak power, and (dis)charge rate.
I would also need to implement continuous evaluation of the SOC.

Would you "recommend" me building on PyBaMM or would a (semi-)empirical model be a better fit for my use case?
Thank you for your time

**Comment by valentinsulzer** [comment, marked as answer]
At this stage PyBaMM can provide you with qualitative insights into the dependence of aging on DoD, peak power, and discharge rate. The results will depend on parameters and model you choose and we can't offer support with selecting these. If you need qualitative insights you can try PyBaMM. If you need fits to data for a specific case, it would be easier with a semi-empirical model
